src/main/ruby/core/rbconfig.rb

```diff
@@ -47,6 +52,8 @@ module RbConfig
   ruby_version   = ruby_api_version
 
   arch         = "#{host_cpu}-#{host_os}"
+  # Host should match the target triplet for clang or GCC, otherwise some C extension builds will fail.
+  host         = "#{host_cpu}-#{host_vendor}-#{host_os}"
```

Comment:
I would move all host-related variables together to clarify how values are used.

Reply:
On 2.3.3 I have:
```
[1] pry(main)> RbConfig::CONFIG["host"]
=> "x86_64-pc-linux-gnu"
```

Reply:
(On 2.2.2 I get `"x86_64-unknown-linux-gnu"`)

Reply:
What do you get if do `gcc -dumpmachine` or `clang --dumpmachine`, if you built those from source?  I think these values are populated in MRI during the build, but there's no good fast way to determine them except by attempting to call the compiler at run time.

Reply:
I get
```
$ gcc -dumpmachine
x86_64-redhat-linux
$ clang -dumpmachine
x86_64-unknown-linux-gnu
```
but that's from my system clang/gcc, I don't compile them manually.

I see, so `unknown` seems fair enough :)